src/Explorer/Notebook/NotebookManager.tsx

```diff
@@ -97,7 +97,9 @@ export default class NotebookManager {
       this.gitHubClient.setToken(token?.access_token);
       if (this?.gitHubOAuthService.isLoggedIn()) {
         useSidePanel.getState().closeSidePanel();
-        this.params.container.openGitHubReposPanel("Manager GitHub settings", this.junoClient);
+        setTimeout(() => {
```

Comment:
Reason why setTimeout is used.

As per earlier implementation, when ever "GitHubReposPanel" KnockOut component was called, always it was loading  as fresh component hence it helps to change the user interface accordingly as per observables.

Since we introduced the Zustand state management, now instead of loading the component from the beginning it is only calling the render method of the "GitHubReposPanel" React component. As we have implemented the same earlier KnockOut logic, our new React component is failing to load when ever state changes in this particular implementation.

Hence the observed solution is, we need to close the panel and provide the required time to settle down the states and re-open the panel again after small internal which  helps to load the fresh React component with required user interface. Therefore, setTimeout is used.